src/auth-helper.js

```diff
@@ -56,31 +67,28 @@ module.exports = {
         if (secondFactorTriggerLogin.status === 429) {
             throw new Error('Rate limit hit... try again later');
         } else if (secondFactorTriggerLogin.status === 401) {
-            console.log(secondFactorTriggerLogin.message);
-            throw new Error("Wrong credentials..." + JSON.stringify(secondFactorTriggerLogin) + secondFactorTriggerLogin.message);
+            throw new Error("Wrong credentials... " + (secondFactorTriggerLogin.message || ''));
         } else if (secondFactorTriggerLogin.status === 400
             && (secondFactorTriggerLoginResult.reason === undefined
                 || secondFactorTriggerLoginResult.reason.missing === undefined)) {
             console.log('something went wrong - got 400 but expected two factor request');
             throw new Error('something went wrong - got 400 but expected two factor request');
-        } else if (secondFactorTriggerLogin.status === 400
-            && (secondFactorTriggerLoginResult.reason !== undefined
-                && !secondFactorTriggerLoginResult.reason.sms)) {
-
-            console.log('Non SMS second factor currently not supported');
-            throw new Error('Non SMS second factor currently not supported');
         } else if (secondFactorTriggerLogin.status !== 400) {
             console.log('something went wrong - expected two factor response but got status' + secondFactorTriggerLogin.status);
         }
 
-        const secondFactor = await read({
+        const secondFactor = secondFactorTriggerLoginResult.reason.phone_number ? await read({
             prompt: 'SMS sent to number ending in (' + secondFactorTriggerLoginResult.reason.phone_number + '): please enter the Authentication code:'
+        }) : await read({
+            prompt: 'Please enter the TOTP authentication code:'
```

Comment:
We check whether there is a status 400 before the prompt and make this dependent on `reason.sms` (see https://github.com/opral/parrot-figcd/pull/5#issuecomment-2256321035).